docs/Usage_example_FDP.ipynb

```diff
@@ -0,0 +1,558 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "8ef630c0-1c7b-4aea-b674-0772d5085aec",
+   "metadata": {},
+   "source": [
+    "# Preparing and uploading data to Fair Data Point with SeMPyRO"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8bff32b5-6f5c-4c6b-953b-e8f9a8ddd681",
+   "metadata": {},
+   "source": [
+    "**Prerequirements:** To execute this notebook in full one needs to have a running Fair Data Point (FDP) instance with an active write access account.\n",
+    "\n",
+    "Let us consider uploading datasets from .example_data.csv to FDP. \n",
+    "FDP requires each dataset to be a part of a catalogue, therefore we need to create a catalogue. Let's see what we need to provide for that:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cff57267-1900-4b21-a8c5-fd5e16c61342",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sempyro.dcat import DCATCatalog\n",
+    "\n",
+    "catalog_fields = DCATCatalog.annotate_model()\n",
+    "print(catalog_fields.mandatory_fields())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9c8b731-d5b8-43aa-9a93-6c4781f51393",
+   "metadata": {},
+   "source": [
+    "Let's create a minimum catalogue with an example title and description. We also need a URI to use as a graph subject at serialization. Let's use `example.com` domain for now for this purpose:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "af6617c9-edef-4b28-a3f5-c36cec1dfa33",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sempyro import LiteralField\n",
+    "from rdflib import URIRef\n",
+    "\n",
+    "catalog_subject = URIRef(\"http://example.com/test_catalog_1\")\n",
+    "\n",
+    "catalog = DCATCatalog(title=[LiteralField(value=\"Test catalog\", language=\"en\")],\n",
+    "                      description=[LiteralField(value=\"Catalog for test example datasets\", language=\"en\")])\n",
+    "catalog_record = catalog.to_graph(catalog_subject)\n",
+    "print(catalog_record.serialize())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "515557de-b2e3-4fb3-a25f-f92b291f1b08",
+   "metadata": {},
+   "source": [
+    "Now we need to define an FDP API client which can log into an FDP, get a token and perform basic calls.\n"
```

Comment:
In an ideal world there's a package for the fdp client, but this probably does not exist yet? Maybe something to consider for Mark's work as well.

Reply:
Agreed on that an FDP client would be desirable.
There is an FDP client from NLeSC which is very similar to what we have, but it's unmaintained (see https://github.com/fair-data/fairdatapoint-client). There's no authentication support, either.

We could consider splitting off our own FDP client into a separate package?

Reply:
imho we should